src/Microsoft.AspNet.Mvc.Core/Logging/MvcRouteHandlerLoggerExtensions.cs

```diff
@@ -36,10 +39,16 @@ public static void ExecutingAction(this ILogger logger, ActionDescriptor action)
             _actionExecuting(logger, action.DisplayName, null);
         }
 
-        public static void ExecutedAction(this ILogger logger, ActionDescriptor action, int startTicks)
+        public static void ExecutedAction(this ILogger logger, ActionDescriptor action, long startTimestamp)
         {
-            var elapsed = new TimeSpan(Environment.TickCount - startTicks);
-            _actionExecuted(logger, action.DisplayName, elapsed.TotalMilliseconds, null);
+            // Don't log if logging wasn't enabled at start of request as time will be wildly wrong.
+            if (logger.IsEnabled(LogLevel.Information) && startTimestamp != 0)
```

Comment:
It should be fine to remove the `logger.IsEnabled(LogLevel.Information)` check since that bit is already embedded in the `startTimestamp != 0` condition. Apart from that, the PR looks good. Presumably we could do something similar with Hosting to avoid the IsEnabled query cost.